### ISSUE TITLE
Fare check index error

### ISSUE THREAD

**Issue by k3vint** [jdholtz/auto-southwest-check-in#224]
### Version

7.2

### Browser Version

121.0.6167.160

### Description

This started happening 3-4 days ago. Fare check causes IndexError. I've tried using account login and individual reservations in config. Also tried on a Linux and Windows machine. Same error every time.

### To Reproduce

Run a fare check, by account login or by reservations

### Relevant logs and program output

```sh
2024-02-14 16:00:19 DEBUG Process-4[reservation_monitor:93]: Checking fares for 1 flights
2024-02-14 16:00:19 DEBUG Process-4[fare_checker:30]: Checking current price for flight
2024-02-14 16:00:19 DEBUG Process-4[fare_checker:88]: Retrieving reservation information
2024-02-14 16:00:21 DEBUG Process-4[utils:32]: Successfully made request after 1 attempts
2024-02-14 16:00:21 DEBUG Process-4[fare_checker:103]: Retrieving search information for the current flight
2024-02-14 16:00:26 DEBUG Process-4[utils:32]: Successfully made request after 1 attempts
2024-02-14 16:00:26 ERROR Process-4[reservation_monitor:106]: Unexpected error during fare check: IndexError('list index out of range')
Traceback (most recent call last):
  File "/home/user/auto-southwest-check-in/lib/reservation_monitor.py", line 100, in _check_flight_fares
    fare_checker.check_flight_price(flight)
  File "/home/user/auto-southwest-check-in/lib/fare_checker.py", line 31, in check_flight_price
    flight_price = self._get_flight_price(flight)
  File "/home/user/auto-southwest-check-in/lib/fare_checker.py", line 49, in _get_flight_price
    flights, fare_type = self._get_matching_flights(flight)
  File "/home/user/auto-southwest-check-in/lib/fare_checker.py", line 80, in _get_matching_flights
    fare_type = fare_type_bounds[bound]["fareProductDetails"]["fareProductId"]
IndexError: list index out of range
2024-02-14 16:00:26 DEBUG Process-4[reservation_monitor:80]: Lock released
```

### Additional context

Linux Mint and Windows 10, without docker

**Comment by jdholtz** [comment]
Thanks for testing this issue out in multiple environments to confirm it is present in every situation. Is the flight a round-trip flight? 

To provide more debugging information, could you add the line `print(change_flight_page, fare_type_bounds)` to line 80 of `lib/fare_checker.py` and paste the output here (removing sensitive information as necessary)?

**Comment by k3vint** [comment]
All of my flights are one-way.

``
2024-02-15 06:16:35 DEBUG Process-4[utils:32]: Successfully made request after 1 attempts
{'messages': [{'key': 'CHANGE_FEE_MESSAGE', 'header': 'none', 'body': "Select the flight(s) you'd like to modify. We never charge change fees. You'll pay only the difference in fare.", 'icon': 'NONE', 'textColor': 'DEFAULT'}], 'dynamicWaivers': [], 'selectionMode': 'ALL', 'boundSelections': [{'flightType': 'Departure', 'originalDate': '2024-04-05', 'fromAirport': 'AAA', 'fromAirportCode': 'BBB', 'toAirport': 'Los Angeles, CA', 'toAirportCode': 'LAX', 'flight': '1111\u200b/\u200b2271', 'timeDeparts': '02:20', 'timeArrives': '13:05', 'showWarningIcon': False, 'isSelectable': True}], 'splitPnrDetails': None, 'passengerDetails': {'disclaimerTextWithLinks': '', 'title': 'PASSENGER(S)', 'passengerList': [{'displayName': 'first last'}]}, '_links': {'changeShopping': {'href': 'v1/mobile-air-booking/page/flights/change/shopping', 'method': 'POST', 'body': [{'boundReference': 'eyJhbGciOiJkaXIiLCJlbmMiLUhTMjU2In0..0YLkbZ1USKinxRuu_TRyQg.mikt0jgweyPXIQg4NmzLxl9C93IUgwmo_5uS8vxkxi1R7GfTjjp-HqxkqVutFOWxcxXBqj4YkgblsscO28UOj1Kp60tLPI4JgNzCzWYy317T392gp5dhsEZKa4yn1NQmpk2Hq0LOa4M8RjgTk6zpnlpqm71_exdG2l4PfZA61nFllF8TYJVcyR04OLF9tSYeS23y-NnZ2kgeRYTrQGku1eWN1o_gvkAszvYE4AWBPEBQ6CWzGOfhDtFMwaiLE5tPMaTwObu7rcYCETx18uLgmNQtUPXCr7UUTPek0f83YQAeon3n_Af4nKSvq2-59UqNbyefltukyyZCj4v7azWtlx7dyig_M0tWk_cem2yi9On1cCKeuKDOGmgywPOUga78Gpue0xE0JxXJ7_nngVQqRay-WdcLE8c4sop9Ovfk_8z74sv1BdOZxSFpN9_iDK-Z3j1VCp66yVO9Cv-S5XHP_XKon7Q10D2NQvG00LLJezm8Br3SdSNSv7UDFGbKA_qmySj9yDLL30XclZgL_qXQom_AdZNOzuQBKrH-9Szznm8oHJ_0eebSpn_hECXPq5gSQvUxG5iywen5Gqd6h6ZOWSBVS6oqvpAykG06mNq8w3TqJJYZLYu8VV96Q-tYqcqnkc4nA2hatT8IW6Io3ohYNZJsFD47e6OmjugAR4MweXjd4hyFxNmMg3ceGSI1wsn2tCOHltKK6eNvfCQq3twfcsysHTO639HaiCXf5pmPWH_tWv7sCHmtbpUGb3aFDWrA872qYa_BzqkMw9g9JXdsqXUUSy13B1CyKUNNSDNLRdkz2bM7Aj_9Vzc3n_UyFuLKiWeakt98AIPLTVaRIFamsNFTB4yFmB4telncVhxjX4zwnfSPJhVr4KRCg9vKNqRC0RL78NmtPOlxU1ayrt2990wJMMmKB5QkwQ_S2t2nqnfSfvyXdnRwku53ixogws_ijg0QnkLK8I2SMLHcsVIpTjafJXmSuSSSChYdXFxvvgtL7wmtVqXgOy9vl8YRyubUo23qfi285Mpbw8I4PnXkJ28m6DnZdcQEp54KuhyXKOHVSPbsluLIOBYzVOmWFsZUeubkUjXyosLb6hYyTbMpcjsOwf2T1PHa5Tz-7Q75D2-cw2N-GjJW5BjmnVGCd6oDgoLrZn2IBiu-751knClwKTrm5_NOENks08MHpi3uGo1aSrfihVhjCR5H0emnsaxIhMXvEyGYH9u3NqbVhVk0fDj0dvSWQNpqXdo_iIbCerQcwGqD2RIKGRvB7g5veOpDdIQILqkUIKzhTO_vH-Sx9qg6sW_ec5bvE4X_86ZBe_6Mcy96yKsOIFX7rGD3OvULvMu_AcRn2i7wI4Cnjvm7DS5f1WUh-0BQAAGzzyxNcBy9RTZWpHYQA_IXaLpJhH1xK8pyCLTEBhvUMTN1hn3ljoQH6OkGSxLx3XSrEePAHYfDknSpHqW4C-j37ipaH7UXK_O5LKVui0fEw1vJWP6K9qY5UIub0h9q06xIDPSOj92rtH4vWVs4ZNTXwT7tByJH2TLRrcJJJSfRrCnCT7DTMOHOxfk1CMGcqHmKsFcxgjtwi3RBevyyCeoIngG2j9DVcP0L6lFT6EhyRXpVFpfS92uvyOGdo4Y74lPmPjHUHo9byXjc3kCNmrKv5EBodqa7hp9iM8y616T0TvIF7UF0O9eAKGwjlPQZMkPAS2MD3gDpFnWvasPGzfSp0MsSgW23PsezrWt--UcgLGlyRcLlmARi8RgqidFDNdjGeWX8zLLkeUPVgxwQEloiIQuZZjYBJxfU6yE9gQFBD_R2dcTqm2EDd8yXQTzh2syQwsURbxW9o54NR77b7RIqEP6KlktTTGYJW7RBQxCUvPvlCqLphyhPgbE5yvw_ayfZ6lrYArKdBQE_kbpMey7g5e4iHcIB_cvAoSZ3d3e7RGEKMbbKxuO_t5IIiRlQhoAGDsnvY3QFqCmipk7oaG0pKcM_UHvu9BURN4Z7CLSJyXQ10xAZzdi77rUkwnru6c8Ki3t-Wag8_MERlRwCEl7fjBlQ9jc6MllOg-Huuf-iVGVu29d-rpl5QwbQ1aEChIz4qfqqQhv80NLabMOTRTyO5MoeAS3TomO2W0-AeIhUMgxr85u57kUzYQ5j62QvdU5JVZvYT8fZz_bMJjUU96PhI6K2sbnlLsW60IweLGn3Cz6CEn4IqgSToKp812LXJnIB9UVkAZCQcr9l95iTVgYDm6RnvCjjvYa9_PRaIcpPw4aBu21SwAghAS21B7DYZdqGQV7THXfXI9T-NFdEfj3gTdG2IByFBwad5XwMSZlvI-d8wp3uExdmWu8FYan4uWxqMTpPQW0T5B4Bpv68XCRe5sflv0P77uBNr8nie-qVoodU5OG4Vx2oDsEanebu39a9gxQbkV1xwfshUnC5E0FsoBHXA1frR5XR51Wt6YTsOPEUJSMyUn9yG95v6bdnSjDRq3NmdNlwroDe5eExOjRPajWJU0KGEF2QNaCGsrhThCXuHpWS2ccca6aF3FzFyRvV47bcTng0AjCqUyYY_w8LBIRNGGsDOFyEpdhd51YEXwSgDQEhQe4JvIjdlnbE07yBY3V4FxsQH6xb3LuqAKc5Q6MDjh86RzmfJava2vx3LZfovcFvyB5oubiglyXawHPX0HuC7mH45t5VR0I5Atqp2sSWcuvZsPo82UfnzKkrMzsrP3J2HeJBhfb90HwTtxvruEN-TrPU3SKdv3GOhU7X58K3sTgV3zTGUq_gv7keP8s5vB6Bak8qk-i_Hut9-2c07JZ_byiZi368j915oTFnVshl-DLN91zNdeXHUpEA7UMtsX1KKpdwyc0G52FRKEubyThIA3sNQFPBqA624EyTxu4s6dDMZkZk1RUg_jO7FlxlaKYHC5Ae667kyfo4oIu4t-sA6g7ewM_BTk5xwDJorEDcIBWyLkC15Zz3GYwt2FRupUJGALNyg1dbdnvcYEakCU18JUV6umPB2X9MLizL1IAGwcAaCKHl733KI1uZ4smxo4fREEXp6AK3KYXNSFCSSQu53DTPRupR1s03Hf_JXr3RynDCHlDnDUmHT-PMUPX9wiPeZd0bnFbB0MMUR496tbsQe_CWFNG8lcNko1Y-jRTof8S8vTbUCFH-2DjVHKD1POtIWtCfv9hO0H5GcanEH6_csgeRQtd2BDmLTVPXb1bhRWAI-vzc35bVqCLqkXYtGIBHP2ElqRbIyyzf6cZ4YXg0rxmCu22Gf3kEWVKz120yPh7bqmmT5HRk-9B9AisSksEEku.xn6cIdQkzH3rLHgdLQuqgA'}]}, 'splitPnr': None}, '_meta': {'hasUnaccompaniedMinor': False, 'isSwabiz': False}, 'mktg_data': {'air_triptype': 'OW', 'air_bound1_changeairportcode': 'AAA:LAX', 'air_bound2_changeairportcode': 'none', 'air_passengercount': '1', 'air_bound1_changedate': '2024-04-05', 'air_bound2_changedate': 'none', 'air_confirmationnumber': 'XXXXXX', 'pnr_split': '0', 'air_daystotrip': '50'}} [{'departureStatus': None, 'departureStatusType': None, 'arrivalStatus': None, 'arrivalStatusType': None, 'flights': [{'number': '1111', 'wifiOnBoard': True, 'aircraftInfo': {'aircraftType': 'Boeing 737 MAX8', 'numberOfSeats': 175, 'wifiSupported': True}}, {'number': '2222', 'wifiOnBoard': True, 'aircraftInfo': {'aircraftType': 'Boeing 737 MAX8', 'numberOfSeats': 175, 'wifiSupported': True}}], 'travelTime': '5h 35m', 'departureDate': '2024-04-05', 'departureTime': '07:20', 'departureAirport': {'name': 'AAA', 'state': 'AA', 'code': 'AAA', 'country': None}, 'arrivalTime': '16:55', 'arrivalAirport': {'name': 'Los Angeles', 'state': 'CA', 'code': 'LAX', 'country': None}, 'passengerTypeCounts': {'adult': 1}, 'passengers': [{'type': 'Passenger', 'count': 1, 'fareType': 'Wanna Get Away'}], 'boundType': 'DEPARTING', 'standbyFlight': None, 'fareProductDetails': {'label': 'Wanna Get Away', 'fareRulesUrl': 'https://mobile.southwest.com/fare-rules/wanna-get-away', 'fareProductId': 'WGARED'}, 'stops': [{'departureStatus': None, 'departureStatusType': None, 'arrivalStatus': None, 'arrivalStatusType': None, 'airport': {'name': 'xxxxxxxx', 'state': 'XX', 'code': 'XXX', 'country': None}, 'arrivalTime': '02:00', 'departureTime': '11:15', 'changePlanes': True, 'missingAirportDetails': False, 'isOvernight': False, 'isNextDayArrival': False}], 'isNextDayArrival': False}]
``

**Comment by camjones1708** [comment]
I'm hitting this as well and believe it's probably due to a one-way flight. I can make that change to test

**Comment by camjones1708** [comment]
I added to the util.make_request
`        elif response.status_code == 400:
            logger.debug(response.json())
`
and received 
`2024-02-15 15:52:55 DEBUG Process-1[utils:35]: {'code': 400310657, 'message': 'Please select the flight(s) you would like to change', 'messageKey': 'VALIDATION__NO_BOUNDS_TO_CHANGE', 'header': None, 'httpStatusCode': 'BAD_REQUEST', 'requestId': 'no-exp-id:0.eaa2c17.1708030374.67c8751:mweb', 'infoList': [{'key': '', 'value': 'Please select the flight(s) you would like to change'}]}
`

I haven't had time to go through all the code, but it looks like the URL is correct. Any thoughts?

**Comment by jdholtz** [comment]
@camjones1708 I am working on this now. The issue seems to be that the flight number from the `fare_type_bounds` never matches the parsed flight number in the `Flight` class, which results in `isChangeBound` being false in the fare check, hence why you are getting an error that says there are no bounds to change.

It seems like Southwest has been refactoring a lot of their mobile website and this issue is a result of one of those effects. In the response k3vint sent, it seems that the flights are separated by `\u200b/\u200b` now instead of just `/`. If you change the following lines in `lib/fare_checker.py`, does the index error disappear?
```diff
--- a/lib/flight.py
+++ b/lib/flight.py
@@ -64,6 +64,6 @@ class Flight:
     def _get_flight_number(self, flights: JSON) -> str:
         flight_number = ""
         for flight in flights:
-            flight_number += flight["number"] + "/"
+            flight_number += flight["number"] + "\u200b/\u200b"
 
-        return flight_number.rstrip("/")
+        return flight_number.rstrip("\u200b/\u200b")
```

**Comment by camjones1708** [comment]
🔥 
`Found lower fare of -50 USD for flight `

**Comment by borski** [comment]
> ```diff
> "\u200b/\u200b"
> ```

This fixed the issue for me as well! :shipit: 🥇

**Comment by jdholtz** [comment]
Thanks for testing everyone! This fix is applied in the develop branch/image

**Comment by k3vint** [comment]
awesome.. works again!
